File: 2019/palindrome/main.java
import java.util.*;
import java.io.*;

public class main {
    public static void main(String[] args) {
        BufferedReader reader;
        String result = "";

        try {
            reader = new BufferedReader(new FileReader("palindrome.in"));
            String line = reader.readLine();

            while (line != null && !line.equals("0 0")) {
                String num = line.split(" ")[0];
                String base = line.split(" ")[1];
                System.out.println("Number is " + num);
                boolean checkIfPalindrome = isPalindrome(convertBase(Integer.parseInt(num),
                        Integer.parseInt(base)));
                String transform = checkIfPalindrome ? "Yes" : "No";
                result += transform + "\n";

                line = reader.readLine();
            }

            FileWriter myWriter = new FileWriter("palindrome.answer");
            myWriter.write(result);

            myWriter.close();
            reader.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static String[] convertBase(int number, int base) {
        String result = "";

        while (number != 0) {
            result += (number % base) + ",";
            number /= base;
        }

        return result.split(",");
    }

    public static boolean isPalindrome(String[] input) {
        System.out.println(Arrays.toString(input));
        boolean result = true;

        for (int i = 0, j = input.length - 1; i < input.length; i++, j--) {
            if (!input[i].equals(input[j])) {
                result = false;
            }
        }

        return result;
    }
}

File: 2019/palindrome/main.py
def transform(num,base):
    final=[]
    if num==0:
        return "0"
    
    while num!=0:
        value=num%base
        final.append(value)
        num=num//base
        
    return final
def check(str):
    if str==str[::-1]:
        return "yes"
    else:
        return "no"
    
    
##################################   main   ########################################
##################################   main   ########################################
##################################   main   ########################################
##################################   main   ########################################
##################################   main   ########################################
##################################   main   ########################################

with open('palindrome.answer', 'w') as file:
    pass

path = "palindrome.in"
lines=[]
with open(path,"r") as file:
     # Iterate through the lines in the file
    for line in file:
        # Strip newline characters and split the line into a list of values
        line_values = line.strip().split()
        
         # Convert the values to integers and append to the 'lines' list
        line_values = [int(value) for value in line_values]
        lines.append(line_values)



#formating
testcase=1
for line in lines:
    
    num=line[0]
    base=line[1]
    
    #perform check
    final=transform(num,base)
    
    #check palindrom
    answer= check(final)
    #answer
    with open('palindrome.answer', 'a') as file:
        # Write the new line to the file
        file.write(str(answer)+"\n")
        
    testcase+=1